### ISSUE TITLE
Color-contrast incorrect stack due to flex children allowing z-index

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3881]
Consider the following HTML:

<details>

```html
<style>
  #shadow {
    display: flex;
  }

  #backdrop {
    position: absolute;
    z-index: 1;
    top: 0;
    left: 0;
    width: 100vw;
    height: 100vh;
    background: rgba(0, 0, 0, 0.5);
  }
</style>

<div id="backdrop"></div>
<div id="shadow"></div>

<script>
  const root = shadow.attachShadow({ mode: 'open' })
  root.innerHTML = `
  <div>asf</div>
  <style>
  div {
    background: white;
    z-index: 1;
  }
  </style>`
</script>
```
</details>

The `div` child of the shadow root is allowed to have a z-index even though it is not using `position`. Turns out that [flex items are allowed to use z-index](https://www.w3.org/TR/css-flexbox-1/#painting) and they are treated as though the element is positioned.

When we [calculate the stacking order for elements](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/dom/create-grid.js#L252-L256), we need to take into account flex children and treat them as though they are positioned. An example of doing that can be found in [the calculation for stacking context](https://github.com/dequelabs/axe-core/blob/develop/lib/commons/dom/create-grid.js#L224-L238).

**Comment by padmavemulapati** [comment]
@straker , can you please some QA notes to validate this ticket?

**Comment by straker** [comment]
For the code sample provided above, running axe should return a passing result (previously it returned incomplete)

**Comment by padmavemulapati** [comment]
Validated with the latest axe-core develop branch code base,
with the test snippet: , now `color-contrast` rule passing (with the latest canary version of axe-4.6.3), earlier it was showing the results in `incompelte` (with the axe-4.4.1)



![Image](https://user-images.githubusercontent.com/47409579/218749664-3d6c7491-25cb-49d0-9390-2149bd63d0d5.png)

Reproducing Screenshot: (with axe-4.4.1)


![Image](https://user-images.githubusercontent.com/47409579/218749975-d47c855c-019a-490b-9582-679250f6e1a9.png)
